trino/sqlalchemy/compiler.py

```diff
@@ -102,6 +112,31 @@ def limit_clause(self, select, **kw):
             text += "\nLIMIT " + self.process(select._limit_clause, **kw)
         return text
 
+    def visit_table(self, table, asfrom=False, iscrud=False, ashint=False,
+                    fromhints=None, use_schema=True, **kwargs):
+        sql = super(TrinoSQLCompiler, self).visit_table(
+            table, asfrom, iscrud, ashint, fromhints, use_schema, **kwargs
+        )
+        return self.add_catalog(sql, table)
+
+    @staticmethod
+    def add_catalog(sql, table):
+        if table is None:
+            return sql
+
+        if isinstance(table, (Alias, CTE, Subquery)):
+            return sql
+
+        if (
+                'trino' not in table.dialect_options
+                or 'catalog' not in table.dialect_options['trino']._non_defaults
+        ):
+            return sql
+
+        catalog = table.dialect_options['trino']._non_defaults['catalog']
```

Comment:
We should avoid to directly call private members of sqlalchemy classes.

I think the following should also work (not tested). 

```suggestion
        catalog = table.dialect_options['trino']['catalog']
```

See https://github.com/sqlalchemy/sqlalchemy/blob/0585c66a20b8075252bb4bbdeef0448f224de96a/lib/sqlalchemy/sql/base.py#L438